test/file_browser.test.js

```diff
@@ -192,4 +203,148 @@ describe("file_browser.html", () => {
+            ]);
+        });
+
+        test("sort by a single column correctly, descending", async () => {
+            // click twice on the "modified" column
+            await click_col_header(1, 2);
+
+            const vals = [];
+            for (let i = 0; i < 10; i++) {
+                const cell_values = await cell_values_at(0, i, 4, 100);
+                vals.push(cell_values[1]);
+            }
+            expect(vals).toEqual([
+                "4/10/1971",
+                "4/9/1971",
+                "4/8/1971",
+                "4/7/1971",
+                "4/6/1971",
+                "4/5/1971",
+                "4/4/1971",
+                "4/3/1971",
+                "4/2/1971",
+                "4/1/1971",
+            ]);
+        });
+
+        test("sort by two columns correctly, [ascending, descending]", async () => {
+            // click one on the "kind" column
+            await click_col_header(2, 1);
+            // shift-click twice on the "modified" column
+            await click_col_header(1, 2, true);
+
+            const vals = [];
+            for (let i = 5; i < 15; i++) {
+                const cell_values = await cell_values_at(0, i, 4, 100);
+                vals.push(cell_values[1]);
+            }
+            expect(vals).toEqual([
+                "1/5/1971",
+                "1/4/1971",
+                "1/3/1971",
+                "1/2/1971",
+                "1/1/1971",
+                "4/10/1971",
+                "4/9/1971",
+                "4/8/1971",
+                "4/7/1971",
+                "4/6/1971",
+            ]);
+        });
+    });
+
+    describe("sorts tree filebrowser", () => {
+        beforeEach(async () => {
+            await refresh_page();
+        });
+
+        test("uses same sort at all levels, sort then expand tree", async () => {
+            // click one on the "kind" column
+            await click_col_header(2, 1);
+            // shift-click twice on the "modified" column
+            await click_col_header(1, 2, true);
+
+            // expand the "easy/" dir, then the "easy/easy" dir
+            await toggle_row_at(5, 4, 100);
+            await toggle_row_at(11, 4, 200);
+
+            const offsets = [200, 106, 12];
+            for (const offset of offsets) {
+                const vals = [];
+                for (let i = 6 + offset; i < 16 + offset; i++) {
+                    const cell_values = await cell_values_at(0, i, 4, 300);
+                    vals.push(cell_values[0]);
+                }
+                expect(vals).toEqual([
+                    "add dog/",
+                    "add charlie/",
+                    "add baker/",
+                    "add able/",
+                    "file_89.txt",
+                    "file_88.txt",
+                    "file_87.txt",
+                    "file_86.txt",
+                    "file_85.txt",
+                    "file_84.txt",
+                ]);
+            }
```

Comment:
This test is valid, but runs into some kind of issue with `scrollTo` when `offset === 12`. Basically, the failure seems to occur due to the fact that `scrollTo(0, 24, 4, 300)` scrolls to the same row as `scrollTo(0, 25, 4, 300)`:

```
  ● file_browser.html › sorts tree filebrowser › uses same sort at all levels, sort then expand tree

    expect(received).toEqual(expected) // deep equality

    - Expected  - 1
    + Received  + 1

    @@ -3,10 +3,10 @@
        "add charlie/",
        "add baker/",
        "add able/",
        "file_89.txt",
        "file_88.txt",
    -   "file_87.txt",
    +   "file_86.txt",
        "file_86.txt",
        "file_85.txt",
        "file_84.txt",
      ]
```